pkg/solana/chain.go

```diff
@@ -557,21 +549,25 @@ func (c *chain) sendTx(ctx context.Context, from, to string, amount *big.Int, ba
 				toKey,
 			).Build(),
 		},
-		blockhash.Value.Blockhash,
+		solana.Hash{}, // Will be set within sendWithRetry txm function.
 		solanago.TransactionPayer(fromKey),
 	)
 	if err != nil {
 		return fmt.Errorf("failed to create tx: %w", err)
 	}
 
-	if balanceCheck {
-		if err = solanaValidateBalance(ctx, reader, fromKey, amountI, tx.Message.ToBase64()); err != nil {
-			return fmt.Errorf("failed to validate balance: %w", err)
-		}
+	msg := &txm.PendingTx{
+		Tx:        *tx,
+		AccountID: "",
+		// To perform balanceCheck we need a blockhash.
+		// Storing values to perform balanceCheck within sendWithRetry txm function before sending tx.
+		BalanceCheck: balanceCheck,
+		From:         fromKey,
+		Amount:       amountI,
 	}
 
 	chainTxm := c.TxManager()
-	err = chainTxm.Enqueue(ctx, "", tx, nil,
+	err = chainTxm.Enqueue(ctx, msg,
```

Comment:
I don't think we should move building the `msg` outside of `Enqueue`. `pendingTx` should just be an internal struct to the TXM for how we want to store/track things. So a caller to `Enqueue` should only worry about the solana tx and pass that in. Then we can build this on our side.